pkgs/applications/window-managers/wtftw/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchgit, rustPlatform, cargo, libXinerama, libX11, xlibs, pkgconfig }:
+
+with import <nixpkgs> { }; with rustPlatform; with xlibs; with stdenv;
+#with rustPlatform;
+
+#with rustPlatform;
+
+buildRustPackage rec {
+  name = "wtftw";
+    src = fetchgit {
+    url = https://github.com/Kintaro/wtftw;
+    rev = "b84fdc04c100ed9d58fbcbd6af072ca553e810f4";
+    sha256 = "09vm2jbdlpamnyg62kzrxcym2cnaz0vcqd3awyiabq507sax6l8l";
+  };
+
+  depsSha256 = "0d0mxbinkryg2rfqwq3p4l49xc7jhrp4bkvxmqz4lr413wvhhbmk";
+
+  buildInputs = [ libXinerama libX11 pkgconfig ];
+  libPath = lib.makeLibraryPath [ libXinerama libX11 ];
+
+  preInstall = ''
+    cargo update
```

Comment:
The build process reveals this:

> > unpacking source archive /nix/store/8i287c95i33japlvj1h9fq0cymqggh39-wtftw-b84fdc0
> source root is wtftw-b84fdc0
> Using cargo deps from /nix/store/naxljsm71ahgbs42mc97baqbxxiv74mh-wtftw-fetch
> Using indexHash '-2304f9dd481834b8'
> Using rust registry from /nix/store/4lgbgn5zc3pxk330k3w9qxa84scs04mk-rustRegistry-2016-04-23-ae2c405
> setting SOURCE_DATE_EPOCH to timestamp 1461705234 of file wtftw-b84fdc0/Cargo.lock
> warning: file wtftw-b84fdc0/Cargo.lock may be generated; SOURCE_DATE_EPOCH may be non-deterministic

However, removing the `cargo update` command gives this error:
>error: some outputs of ‘/nix/store/5r5p4hzb54bxzsk3i4cr13l72rwk437p-wtftw.drv’ are not valid, so checking is not possible